core/src/test/java/org/elasticsearch/cluster/routing/RoutingTableGenerator.java

```diff
@@ -111,6 +118,12 @@ public void update(ShardRouting shardRouting) {
 
             if (shardRouting.primary()) {
                 primaryInactive++;
+                final UnassignedInfo unassignedInfo = shardRouting.unassignedInfo();
+                assert unassignedInfo != null;
+                if (unassignedInfo.getReason() != UnassignedInfo.Reason.INDEX_CREATED
```

Comment:
this is essentially the same code as in ClusterHealth. Factor this into  a method?